update RN to apply XCode 14.3 patch:

PoorTable opened the pull request:
bump ruby, adjust gemfile. adjust AppDelegate to make splash screen work after latest updates

PoorTable committed:
update RN to apply XCode 14.3 patch:
bump ruby, adjust gemfile. adjust AppDelegate to make splash screen work after latest updates

sharnik commented on template/.ruby-version:
```diff
@@ -1 +0,0 @@
-2.7.5
```
We still need this file to stay in sync inside the team which Ruby version we're all using.

sharnik commented on template/Gemfile:
```diff
@@ -1,4 +1,5 @@
 source 'https://rubygems.org'
 # You may use http://rbenv.org/ or https://rvm.io/ to install and use this version
-ruby File.read(File.join(__dir__, '.ruby-version')).strip
+ruby '>= 2.6.10'
```
Let's pick a particular version here, because we need to pick one for the .ruby-version file anyway. I'd use 2.7.6, because Facebook people recommend it in the GH issues sometimes.

PoorTable replied:
That's how done in 0.71.6 RN default template, they deleted `.ruby-version`, and put this version definition here, but yep, technically make sense to bring  `.ruby-version` back and put 2.7.6, I like this version too

PoorTable committed:
put specific version of ruby in .ruby-version

sharnik reviewed (changes requested):
When you boot up the app, it seems the API_URL from Config (https://github.com/brains-and-beards/react-native-template-redbeard/blob/develop/template/src/api/common.ts#L10) is not picked up correctly (it's `undefined`) and we get networking errors.


![CleanShot 2023-04-14 at 10 41 26@2x](https://user-images.githubusercontent.com/987/231992218-bcf75d59-02ab-457d-b284-26557fcb8a78.png)

PoorTable committed:
add example env to main project

PoorTable commented:
> When you boot up the app, it seems the API_URL from Config (https://github.com/brains-and-beards/react-native-template-redbeard/blob/develop/template/src/api/common.ts#L10) is not picked up correctly (it's `undefined`) and we get networking errors.
> 
> ![CleanShot 2023-04-14 at 10 41 26@2x](https://user-images.githubusercontent.com/987/231992218-bcf75d59-02ab-457d-b284-26557fcb8a78.png)

It was never used at all, fixed, added pre-build action that uses it

sharnik reviewed (commented):
I was thinking a bit about this fix and I don't like two things here:
1. It's iOS only
2. We're programming a workaround for people (like me) who didn't read the README. 

Basically point 2 means a fix is not necessary. If I had read the README, I'd set up the app correctly 🤷‍♂️
Let's remove the last commit here (`add example env to main project `) and merge it 🚀

PoorTable committed:
Revert "add example env to main project"

This reverts commit 18a03d31837d65657af43621d1136c017ab8ed5c.

PoorTable commented:
> I was thinking a bit about this fix and I don't like two things here:
> 
> 1. It's iOS only
> 2. We're programming a workaround for people (like me) who didn't read the README.
> 
> Basically point 2 means a fix is not necessary. If I had read the README, I'd set up the app correctly 🤷‍♂️ Let's remove the last commit here (`add example env to main project `) and merge it 🚀

Reverted :)
